scripts/fetch-agent-releases.js

```diff
@@ -0,0 +1,617 @@
+ */
+async function fetchReleases(owner, repo) {
+  const releases = [];
+  let page = 1;
+  const perPage = 100;
+
+  while (true) {
+    try {
+      const path = `/repos/${owner}/${repo}/releases?per_page=${perPage}&page=${page}`;
+      console.log(`Fetching releases from ${owner}/${repo} (page ${page})...`);
+      
+      const pageReleases = await githubRequest(path);
+      
+      if (pageReleases.length === 0) {
+        break;
+      }
+      
+      releases.push(...pageReleases);
+      
+      if (pageReleases.length < perPage) {
+        break;
+      }
+      
+      page++;
+    } catch (error) {
+      console.error(`Error fetching releases for ${owner}/${repo}: ${error.message}`);
+      break;
+    }
+  }
+
+  return releases;
+}
+
+/**
+ * Normalize version by removing leading 'v'
+ */
+function normalizeVersion(version) {
+  return version.replace(/^v/, '');
+}
+
+/**
+ * Determine release type from version string
+ */
+function getReleaseType(version) {
+  const normalized = version.toLowerCase();
+  
+  if (normalized.includes('-rc') || normalized.includes('.rc') || normalized.includes('+rc') || 
+      normalized.includes('release-candidate') || normalized.includes('-preview') || normalized.includes('+preview')) {
+    return 'release-candidate';
+  }
+  if (normalized.includes('-beta') || normalized.includes('.beta') || normalized.includes('+beta') || 
+      normalized.includes('-swift') || normalized.includes('+swift')) {
+    return 'beta';
+  }
+  if (normalized.includes('-alpha') || normalized.includes('.alpha') || normalized.includes('+alpha') || 
+      normalized.includes('-experiment') || normalized.includes('+experiment') || 
+      normalized.includes('-dev') || normalized.includes('+dev')) {
+    return 'alpha';
+  }
+  
+  // Check for any other suffix
+  if (normalized.includes('-') || normalized.includes('+')) {
+    const suffix = normalized.split(/[-+]/)[1];
+    if (suffix) {
+      console.warn(`  ⚠️  Unknown version suffix in ${version}, categorizing as alpha`);
+      return 'alpha';
+    }
+  }
+  
+  return 'stable';
+}
+
+/**
+ * Compare semantic versions
+ */
+function compareVersions(a, b) {
+  const parseVersion = (v) => {
+    const clean = normalizeVersion(v);
+    const parts = clean.split(/[-+]/)[0].split('.');
+    return parts.map(p => parseInt(p) || 0);
+  };
+
+  const aParts = parseVersion(a);
+  const bParts = parseVersion(b);
+
+  for (let i = 0; i < Math.max(aParts.length, bParts.length); i++) {
+    const aPart = aParts[i] || 0;
+    const bPart = bParts[i] || 0;
+    
+    if (aPart > bPart) return 1;
+    if (aPart < bPart) return -1;
+  }
+  
+  return 0;
+}
+
+/**
+ * Check if a version is less than or equal to a threshold
+ */
+function isVersionLessThanOrEqual(version, threshold) {
+  return compareVersions(version, threshold) <= 0;
+}
+
+/**
+ * Calculate support end date (12 months after release)
+ */
+function calculateSupportEndDate(releaseDate) {
```

Comment:
@mattheworiordan I think this also needs to be based on the superseding release date.